Data/Scripts/GardenConquest/Extensions/GridExtensions.cs

```diff
@@ -184,6 +186,48 @@ public static class GridExtensions {
 			return false;
 		}
 
+		/// <summary>
+		/// Get the classifier block on the grid
+		/// </summary>
+		/// <param name="grid"></param>
+		/// <remarks>Grids should only have 1 classifier block</remarks>
+		/// <returns>The HullClassifier as IMyCubeBlock if found, null otherwise</returns>
+		public static IMyCubeBlock getClassifierBlock(this IMyCubeGrid grid) {
+			List<IMySlimBlock> beaconBlocks = new List<IMySlimBlock>();
+
+			// Get all beacon blocks
+			Func<IMySlimBlock, bool> isBeacon = b => b.FatBlock != null && b.FatBlock is InGame.IMyBeacon;
+			grid.GetBlocks(beaconBlocks, isBeacon);
```

Comment:
Would you mind defining isBeacon inline within the call to grid.GetBlocks? Either that, or assign it outside of this function, so that we don't have to do new object allocation every time for this.